aloha-core/src/main/scala/com/eharmony/aloha/models/multilabel/MultilabelModel.scala

```diff
@@ -0,0 +1,387 @@
+  *                          produced by predictorProducer is Closeable, it will be closed when
+  *                          MultilabelModel's close method is called.
+  * @param numMissingThreshold if provided, we check whether the threshold is exceeded.  If so,
+  *                            return an error instead of the computed score.  This is for missing
+  *                            data situations.
+  * @param auditor transforms a `Map[K, Double]` to a `B`.  Reports successes and errors.
+  * @tparam U upper bound on model output type `B`
+  * @tparam K type of label or class
+  * @tparam A input type of the model
+  * @tparam B output type of the model.
+  */
+case class MultilabelModel[U, K, -A, +B <: U](
+    modelId: ModelIdentity,
+    featureNames: sci.IndexedSeq[String],
+    featureFunctions: sci.IndexedSeq[GenAggFunc[A, Sparse]],
+    labelsInTrainingSet: sci.IndexedSeq[K],
+    labelsOfInterest: Option[GenAggFunc[A, sci.IndexedSeq[K]]],
+    predictorProducer: SparsePredictorProducer[K],
+    numMissingThreshold: Option[Int],
+    auditor: Auditor[U, Map[K, Double], B])
+extends SubmodelBase[U, Map[K, Double], A, B]
+   with RegressionFeatures[A] {
+
+  import MultilabelModel._
+
+  /**
+    * predictory is transient lazy value because we don't need to worry about serialization.
+    * We don't care about the lazy property.  It should be created eagerly.
+    */
+  @transient private[this] lazy val predictor = predictorProducer()
+  predictor // Force predictor eagerly
+
+  /**
+    * Cache this in case labelsOfInterest is None.  In that case, we don't want to repeatedly
+    * create this because it could create a GC burden for no real reason.
+    */
+  @transient private[this] lazy val defaultLabelInfo =
+    LabelsAndInfo(labelsInTrainingSet.indices, labelsInTrainingSet, Seq.empty, None)
+
+  /**
+    * Making from label to index into the sequence of all labels encountered during training.
+    */
+  private[this] val labelToInd: Map[K, Int] =
+    labelsInTrainingSet.zipWithIndex.map { case (label, i) => label -> i }(collection.breakOut)
+
+  override def subvalue(a: A): Subvalue[B, Map[K, Double]] = {
+    val li = labelsAndInfo(a, labelsOfInterest, labelToInd, defaultLabelInfo)
+
+    if (li.labels.isEmpty)
+      reportNoPrediction(modelId, li, auditor)
+    else {
+      val Features(x, missing, missingOk) = constructFeatures(a)
+
+      if (!missingOk)
+        reportTooManyMissing(modelId, li, missing, auditor)
+      else {
+        // TODO: To support label-dependent features, fill last parameter with a valid value.
+        val predictionTry = Try { predictor(x, li.labels, li.indices, sci.IndexedSeq.empty) }
+
+        predictionTry match {
+          case Success(pred) => reportSuccess(modelId, li, missing, pred, auditor)
+          case Failure(ex)   => reportPredictorError(modelId, li, missing, ex, auditor)
+        }
+      }
+    }
+  }
+
+  override def close(): Unit =
```

Comment:
Probably add a comment about what is closed and when.

Reply:
I don't think it's necessary.

Reply:
Did it anyway.